### DISCUSSION TITLE
Security issues

### DISCUSSION THREAD

**Discussion by HelloWorldOmega** [codetheweb/muse#929] [General]
Hi, i run muse correctly on a synology with docker as a test.
Then i wanted to deploy it into my work environment because of i have connection issues at my home but it seems the firewall is blocking the network dedicated to the container so when it tries to do the job it doesn't work and it crashes continuosly.
After a bit of tries i isolated the network of muse in a separate VLAN and set firewall rule to let pass every traffic on the ip address of the container but i don't have time to check ports and connection logs to find on which ports it works, so my question is:
what ports are in use by services inside so i can block every ports not necessary?

**Comment by codetheweb** [comment]
Muse doesn't use any special ports besides what discord.js uses.

This may be helpful: https://github.com/discordjs/discord.js/issues/8728

**Reply by GrantBirki** [reply]
@codetheweb I'm kind of confused here.. It looks like other users are reporting networking / port issues as well but I don't understand why ([ref](https://github.com/codetheweb/muse/issues/955#issuecomment-1689328765)). As far as I know, Discord Bots don't need any open ports to operate. However, there is certainly something up with the way my bot is running.

My bot is running Docker (docker compose) ([source code](https://github.com/GrantBirki/muse/blob/main/docker-compose.yml)) on a Linux machine. When using my personal Linux desktop, the bot works completely fine. However, when I deploy the _exact same code_ on my production Linux VPS (still using docker + docker compose), the bot completely fails to play any audio over Discord.

I have tried two separate open source bot project now (first started with [evobot](https://github.com/eritislami/evobot)), and both have this exact same issue.

If you have any ideas or ways that I could troubleshoot this, I would be very grateful! It seems like other users are having this problem as well and point out "mysterious port issues" but I'm not exactly sure _how_ that would be the case. Thanks! :bow:

**Reply by GrantBirki** [reply]
Update: It was indeed somewhat related to my cloud provider's network layer firewall. I fixed it and documented what was happening [here](https://github.com/GrantBirki/muse/blob/6c1575be84732419e702e719730cb9b511c62e0c/docs/ovh.md)

**Reply by codetheweb** [reply]
very interesting, thanks for the follow up

would be happy to add this to Muse's docs if you're interested in making a PR :)
